Dockerfiles/agent-ot/Dockerfile.byoc

```diff
@@ -0,0 +1,82 @@
+# Use the Ubuntu Slim AMD64 base image
+FROM ubuntu:24.04 AS builder
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV GO_VERSION=1.22.0
+ENV AGENT_VERSION=7.56.0-rc.7
```

Comment:
I would not hardcode here this; instead I would default to `main` and allow overriding of the AGENT_VERSION with a [build argument](https://docs.docker.com/build/guide/build-args/).

Reply:
Same for `GO_VERSION` let's allow overriding with a build argument.

Reply:
+1

Reply:
done